usaspending_api/references/v2/views/filter_tree/tas_filter_tree.py

```diff
@@ -0,0 +1,100 @@
+from usaspending_api.common.helpers.agency_logic_helpers import cfo_presentation_order
+from usaspending_api.accounts.models import TreasuryAppropriationAccount, FederalAccount
+from usaspending_api.awards.models.financial_accounts_by_awards import FinancialAccountsByAwards
+from usaspending_api.references.v2.views.filter_tree.filter_tree import DEFAULT_CHILDREN, Node, FilterTree
+from usaspending_api.references.models import ToptierAgency
+from django.db.models import Exists, OuterRef
+
+
+class TASFilterTree(FilterTree):
+    def toptier_search(self):
+        agency_set = (
+            ToptierAgency.objects.annotate(
+                has_faba=Exists(
+                    FinancialAccountsByAwards.objects.filter(
+                        treasury_account__federal_account__parent_toptier_agency=OuterRef("pk")
```

Comment:
This is my fault.  Since we're applying this to advanced search, we really want to check if there's File D data.  This is checking for File C only.  We need to take it one step further.

To this filter function (and the others like it), can we add  `award_id__isnull=False'?

Sorry about that.

Reply:
I've trusted you this far...

Reply:
aaaand look where that's gotten you...

Reply:
walked right into that one